### PR TITLE
HW 6: Task 2

### PR DESCRIPTION
**Pull request by ValeriL**

### CODE REVIEW

File: Homework/homework6/test_task2.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+    Cls()
+
+    assert Cls.instances_counter == 1
+
+
+def test_get_created_instances():
+    @instances_counter
+    class Cls:
+        pass
```
Comment (Cjkjvfnby) [inline Homework/homework6/test_task2.py:17]: We can decorate any class. 
Please use this.

```python
class Cls:
    instances_counter = 42
```

File: Homework/homework6/test_task2.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+
+def test_instances_are_counted():
+    @instances_counter
+    class Cls:
+        pass
+
+    Cls()
+
+    assert Cls.instances_counter == 1
```
Comment (Cjkjvfnby) [inline Homework/homework6/test_task2.py:11]: Which additional value this tests add to the `test_get_created_instances`?

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]

File: Homework/homework6/test_task2.py (hunk 2 above)
Comment (ValeriL) [reply Homework/homework6/test_task2.py:11]: It was a logic behind that if the variable changes right here but test `test_get_created_instances` fails than smth wrong with the function... But okay, I see, maybe such situation is not likely for such function.

File: Homework/homework6/test_task2.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+def test_get_created_instances():
+    @instances_counter
+    class Cls:
+        instances_counter = 42
+
+    Cls()
+
+    assert Cls.get_created_instances() == 1
+    assert Cls.instances_counter == 42
```
Comment (Cjkjvfnby) [inline Homework/homework6/test_task2.py:12]: Now you just test a random attribute.  Using `__instances_counter` as the name has more sense.

**Review by Cjkjvfnby — APPROVED** [review]

**Pull request merged**